runtime/src/iree/builtins/ukernel/arch/arm_64/mmt4d_arm_64_base.c

```diff
@@ -264,3 +264,118 @@ IREE_UK_MMT4D_TILE_FUNC_IMPL_FOR_M0_1_2_4_8(
+    int16x8_t rhs_high_0 = vmovl_s8(vget_low_s8(rhs_high));
+    int16x8_t rhs_high_1 = vmovl_s8(vget_high_s8(rhs_high));
+
+    acc[0] = vmlal_n_s16(acc[0], vget_low_s16(rhs_low_0), lhs_0);
+    acc[1] = vmlal_n_s16(acc[1], vget_high_s16(rhs_low_0), lhs_0);
+    acc[2] = vmlal_n_s16(acc[2], vget_low_s16(rhs_low_1), lhs_0);
+    acc[3] = vmlal_n_s16(acc[3], vget_high_s16(rhs_low_1), lhs_0);
+
+    acc[0] = vmlal_n_s16(acc[0], vget_low_s16(rhs_high_0), lhs_1);
+    acc[1] = vmlal_n_s16(acc[1], vget_high_s16(rhs_high_0), lhs_1);
+    acc[2] = vmlal_n_s16(acc[2], vget_low_s16(rhs_high_1), lhs_1);
+    acc[3] = vmlal_n_s16(acc[3], vget_high_s16(rhs_high_1), lhs_1);
+
+    if (M0 >= 2) {
+      int16_t lhs_2 = *lhs_ptr++;
+      int16_t lhs_3 = *lhs_ptr++;
+
+      // Handle 2x16x2.
+      acc[4] = vmlal_n_s16(acc[4], vget_low_s16(rhs_low_0), lhs_2);
+      acc[5] = vmlal_n_s16(acc[5], vget_high_s16(rhs_low_0), lhs_2);
+      acc[6] = vmlal_n_s16(acc[6], vget_low_s16(rhs_low_1), lhs_2);
+      acc[7] = vmlal_n_s16(acc[7], vget_high_s16(rhs_low_1), lhs_2);
+
+      acc[4] = vmlal_n_s16(acc[4], vget_low_s16(rhs_high_0), lhs_3);
+      acc[5] = vmlal_n_s16(acc[5], vget_high_s16(rhs_high_0), lhs_3);
+      acc[6] = vmlal_n_s16(acc[6], vget_low_s16(rhs_high_1), lhs_3);
+      acc[7] = vmlal_n_s16(acc[7], vget_high_s16(rhs_high_1), lhs_3);
+
+      if (M0 == 4) {
+        int16_t lhs_4 = *lhs_ptr++;
+        int16_t lhs_5 = *lhs_ptr++;
+        int16_t lhs_6 = *lhs_ptr++;
+        int16_t lhs_7 = *lhs_ptr++;
+
+        // Handle 4x16x2.
+        acc[8] = vmlal_n_s16(acc[8], vget_low_s16(rhs_low_0), lhs_4);
+        acc[9] = vmlal_n_s16(acc[9], vget_high_s16(rhs_low_0), lhs_4);
+        acc[10] = vmlal_n_s16(acc[10], vget_low_s16(rhs_low_1), lhs_4);
+        acc[11] = vmlal_n_s16(acc[11], vget_high_s16(rhs_low_1), lhs_4);
+
+        acc[8] = vmlal_n_s16(acc[8], vget_low_s16(rhs_high_0), lhs_5);
+        acc[9] = vmlal_n_s16(acc[9], vget_high_s16(rhs_high_0), lhs_5);
+        acc[10] = vmlal_n_s16(acc[10], vget_low_s16(rhs_high_1), lhs_5);
+        acc[11] = vmlal_n_s16(acc[11], vget_high_s16(rhs_high_1), lhs_5);
+
+        acc[12] = vmlal_n_s16(acc[12], vget_low_s16(rhs_low_0), lhs_6);
+        acc[13] = vmlal_n_s16(acc[13], vget_high_s16(rhs_low_0), lhs_6);
+        acc[14] = vmlal_n_s16(acc[14], vget_low_s16(rhs_low_1), lhs_6);
+        acc[15] = vmlal_n_s16(acc[15], vget_high_s16(rhs_low_1), lhs_6);
+
+        acc[12] = vmlal_n_s16(acc[12], vget_low_s16(rhs_high_0), lhs_7);
+        acc[13] = vmlal_n_s16(acc[13], vget_high_s16(rhs_high_0), lhs_7);
+        acc[14] = vmlal_n_s16(acc[14], vget_low_s16(rhs_high_1), lhs_7);
+        acc[15] = vmlal_n_s16(acc[15], vget_high_s16(rhs_high_1), lhs_7);
+      }
+    }
+  }
+
+  for (int i = 0; i < 4 * M0; ++i) {
+    // Divide by 16 since we shifted int4s to the upper 4 bits of int8s.
+    acc[i] = vshrq_n_s32(acc[i], 4);
+    // Add existing values after scaling back by x16.
```

Comment:
maybe that comment doesn't add more useful info on top of what the comment 2 lines above provided.

Reply:
Removed